### PR TITLE
Adds functional tests for /plugincheck

### PR DESCRIPTION
**Pull request by schalkneethling**
@alexgibson r?

### CODE REVIEW

File: tests/functional/plugincheck.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    casper.userAgent('Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:38.0) Gecko/20100101 Firefox/38.0');
+
+    casper.thenOpen(url, function() {
+
+        test.assertHttpStatus(200);
+
+        test.assertVisible('.plugin-status-container h2', 'Plugin Status heading exists');
+        test.assertNotVisible('.not-supported', 'Unsupported browser message not visible');
+
+        casper.userAgent('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36');
```
Comment (alexgibson) [inline tests/functional/plugincheck.js]: Is this resetting the US back to the default? If so, would it make sense to do this automatically after every test in the `done()` function in `/lib/helpers.js`?

File: tests/functional/plugincheck.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    });
+
+    casper.thenOpen(url, function() {
+
+        test.assertHttpStatus(200);
```
Comment (alexgibson) [inline tests/functional/plugincheck.js]: Probably don't need to check the response code in every test? In other examples, I've just done this once as the first test in every suite

File: tests/functional/plugincheck.js (hunk 1 above)
Comment (alexgibson) [inline tests/functional/plugincheck.js]: Never mind, this gets reset after each test block :+1:

File: tests/functional/plugincheck.js (hunk 2 above)
Comment (schalkneethling) [inline tests/functional/plugincheck.js]: I was wondering about that. I was thinking that each block will reload the page so, perhaps just ensuring it actually loaded before stepping into the real tests might be a good 'sanity' check but, no reason to overdo things.

You reckon I should remove?

Comment (alexgibson) [inline tests/functional/plugincheck.js]: I think once should be enough, since using it everywhere quickly runs up the number of total tests?

Comment (schalkneethling) [inline tests/functional/plugincheck.js]: I am wondering though, perhaps for the two tests where the userAgent is changed, perhaps here it makes sense to do a status check inside `.thenOpen`?

File: tests/functional/plugincheck.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+casper.test.begin('PluginCheck, Elements' + url, 4, function suite(test) {
+
+    casper.start();
+
+    casper.userAgent(config.userAgent.fx);
+
+    casper.thenOpen(url, function() {
+        test.assertHttpStatus(200);
+        test.assertResourceExists(/media\/img\/plugincheck\/screenshots\/addons-screenshot(\.\w{1,}){0,}\.jpg/);
+        // The tests below ensure that the newsletter section is always visible irrespective of browser
```
Comment (alexgibson) [inline tests/functional/plugincheck.js]: Is there any logic in the page to hide the newsletter? If not, is the extra test below necessary?

File: tests/functional/plugincheck.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('PluginCheck, Download button present: ' + url, 1, function suite(test) {
+
+    casper.start(url, function() {
+        test.assertVisible('#download-button-desktop-release', 'Download button visible');
```
Comment (alexgibson) [inline tests/functional/plugincheck.js]: Could this be moved up into the first test?

File: tests/functional/plugincheck.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+var config = require('../lib/config');
+var helpers = require('../lib/helpers');
+var path = '/plugincheck/';
+var url = config.base() + path;
+
+casper.options.pageSettings.loadPlugins = true;
+
+
+casper.test.begin('PluginCheck, Elements' + url, 4, function suite(test) {
```
Comment (alexgibson) [inline tests/functional/plugincheck.js]: Nit: need a space after `Elements`

File: tests/functional/plugincheck.js (hunk 3 above)
Comment (schalkneethling) [inline tests/functional/plugincheck.js]: No, I guess if shows in the one, the other will be fine as well.

File: tests/functional/plugincheck.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+/* global casper */
+'use strict';
+
+var config = require('../lib/config');
+var helpers = require('../lib/helpers');
+var path = '/plugincheck/';
+var url = config.base() + path;
+
+casper.options.pageSettings.loadPlugins = true;
```
Comment (alexgibson) [inline tests/functional/plugincheck.js]: Is this needed?

File: tests/functional/plugincheck.js (hunk 3 above)
Comment (schalkneethling) [inline tests/functional/plugincheck.js]: Then again, seeing I am moving https://github.com/mozilla/bedrock/pull/3048#discussion-diff-32004107R84 up into this test, no harm in keeping the additional assert is there?

File: tests/functional/plugincheck.js (hunk 4 above)
Comment (schalkneethling) [inline tests/functional/plugincheck.js]: Yup, moving.

File: tests/functional/plugincheck.js (hunk 6 above)
Comment (schalkneethling) [inline tests/functional/plugincheck.js]: Nah, it is going to have no effect, removing.

File: tests/functional/plugincheck.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+casper.test.begin('PluginCheck, Elements' + url, 4, function suite(test) {
+
+    casper.start();
+
+    casper.userAgent(config.userAgent.fx);
+
+    casper.thenOpen(url, function() {
+        test.assertHttpStatus(200);
+        test.assertResourceExists(/media\/img\/plugincheck\/screenshots\/addons-screenshot(\.\w{1,}){0,}\.jpg/);
```
Comment (alexgibson) [inline tests/functional/plugincheck.js]: Could this be simplified to just `test.assertResourceExists(/media\/img\/plugincheck\/screenshots\/addons-screenshot/);` maybe - thoughts?

Comment (schalkneethling) [inline tests/functional/plugincheck.js]: probably, I just tested it directly in a Regexp tester thingy and it seemed good and stable, but I reckon it does not matter what the file type is, nor those numbers although, perhaps the function will care.

Comment (schalkneethling) [inline tests/functional/plugincheck.js]: Seems it is totally happy with that :cake:

**Comment by alexgibson** [comment]
r+ :banana:

**Pull request merged**